### DISCUSSION TITLE
Electron User Agent is missing

### DISCUSSION THREAD

**Discussion by tobeagram** [plausible/analytics#1097] [Anything Else]
## Prerequisites
- [X] I have searched open and closed issues to make sure that the bug has not yet been reported.

## Bug report
`Electron` user agent (UA) string is not picked up by Plausible, and is omitted from `client_browsers.yml` and `short_codes.client_browsers.yml` files. I have previously opened this as a [feature request](https://github.com/plausible/analytics/discussions/1059), but after more careful consideration find this to be a bug rather than an FR, since `Electron` is a recognised UA by the majority of other services, including Sentry. 

**Expected behavior**
Add `Electron` UA to client_browsers.

**Comment by metmarkosaric** [comment]
thanks @tobeagram! we keep issues to actual bugs only and this is more like a feature request so moving it here and we can consider it.

**Comment by ukutaht** [comment]
We inherit the device detection behaviour from https://github.com/matomo-org/device-detector/. It's probably the most popular user-agent parsing library our there so it would interesting to find out why they don't include Electron as a separate browser.

**Comment by tobeagram** [comment]
Thanks for your replies @metmarkosaric & @ukutaht! Very strange that Matomo doesn't include Electron as a separate browser in their DeviceDetector library; there's [an out of date closed issue from 2017](https://github.com/matomo-org/device-detector/issues/5597), explaining that Electron is a framework; but I think that it's definitely a lot more useful to distinguish clients as `Electron` rather than `Chrome` from an analytics standpoint, like Plausible. 

The more popular UA Parser, https://github.com/faisalman/ua-parser-js, lists `Electron` as a browser as well. @ukutaht what are your thoughts?

**Reply by ukutaht** [reply]
I tend to agree with you @tobeagram. I think Electron could and should be classified as a browser in its own right. There are tons of other Chromium-based browsers that do get their own entry, why not Electron?

However, I don't think switching libraries is a good option at this point, it would probably be easier to just fork device-detector if we need it. Or we could even attempt to add it in a PR and see if it gets accepted or shot down.

Are you looking for this feature to track how many people are using a desktop app vs a browser?

**Reply by tobeagram** [reply]
Yes exactly, we would like to track how many people are using a desktop app vs a browser. At the moment everyone using 'Chrome' and 'Desktop Apps' are lumped into `Chrome` under 'Devices > Browser' in Plausible, which isn't helpful and is skewing the actual amount of people using Chrome. 

A huge benefit of having `Electron` as a separate browser is we can then use Plausible's filtering to understand which 'Operating Systems', 'Formats' and 'Countries' are most popular on Desktop Apps.

I also completely understand that switching libraries is not viable option – attempting a PR and then forking DeviceDetector in the meantime sounds like a great option for sure.
